### PR TITLE
Shreyansh | Week4 - Implemented routing and Integrated it with main project

### PR DESCRIPTION
**Pull request by shreyansh-khatri**
1) Briefly describe your change.
 Uploaded week4 assignment of ReactJS and also integrated it in the common project for all trainees.
 
 2) What parts of the application are affected by your change?
 My changes only affected the week4-Shreyansh branch and the components inside it.
 
 3) How did you test your change?
 I tested various parts such as validation, different inputs and some other edge cases.

### CODE REVIEW

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3Style.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { styled } from "styled-components";
+
+export const Feature3Wrapper = styled.div`
+  .currencyConverterContainer {
+    max-width: 400px;
+    margin: 0 auto;
+    padding: 20px;
+    background-color: #ffffff;
+    border-radius: 10px;
+    box-shadow: 0 0 5px rgba(0, 0, 0, 0.1);
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3Style.ts]: use single common way of styling colors

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/index.tsx [hunk 2]
```diff
@@ -1,12 +1,26 @@
-import React from 'react'
-import { ShreyanshWrapper } from './styles'
+import React, { useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
+import { ShreyanshWrapper } from './styles';
 
 const Shreyansh = () => {
+  const [selectedWeek, setSelectedWeek] = useState('');
```
Comment (shreegovinda) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/index.tsx]: type

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/DashboardStyle.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { styled } from "styled-components";
+
+export const DashboardWrapper = styled.div`
+  .dashboard-container {
+    max-width: 800px;
+    margin: 30px auto 0;
+    padding: 20px;
+    background-color: gray;
```
Comment (Harrrrry) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/DashboardStyle.ts]: use hexacode

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature2/Feature2.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import { Feature2Wrapper } from "./Feature2Style";
```
Comment (Harrrrry) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature2/Feature2.tsx]: I recommend to have proper name of file and folder.
instead of feature2, it should be calculator

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            id="currencyInput"
+            type="number"
+            value={amount}
+            onChange={handleAmountChange}
+          />
+        </div>
+        <div>
+          <label>From:</label>
+          <select value={baseCurrency} onChange={handleBaseCurrencyChange}>
+            <option value="USD">USD</option>
```
Comment (Harrrrry) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx]: Use labels for string

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            <option value="EUR">EUR</option>
+            <option value="INR">INR</option>
+          </select>
+        </div>
+        <div>
+          <label>To:</label>
+          <select value={targetCurrency} onChange={handleTargetCurrencyChange}>
```
Comment (Harrrrry) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx]: This select is duplicate code, It should be common with params as per the functionality

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/index.tsx [hunk 7]
```diff
@@ -1,12 +1,26 @@
...
+
+  const handleWeekClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    setSelectedWeek(e.currentTarget.value);
+  };
+
   return (
     <ShreyanshWrapper>
```
Comment (Harrrrry) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/index.tsx]: Use proper name

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/styles.ts [hunk 8]
```diff
@@ -1,5 +1,38 @@
 import styled from "styled-components";
 
 export const ShreyanshWrapper = styled.div`
-    
+       .weekButtonsContainer {
```
Comment (Harrrrry) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/styles.ts]: Do proper indentation

**Review by Harrrrry — CHANGES_REQUESTED** [review]

File: interns-react-weekly-assignments/src/App.tsx [hunk 9]
```diff
@@ -10,7 +10,7 @@ import NotFound from './components/NotFound';
 import Week1 from './components/Home/Keshav/Week1';
 import Week3 from './components/Home/Keshav/Week3';
 import Week4 from './components/Home/Keshav/Week4';
-
+import Week_4 from './components/Home/Shreyansh/Week_4'
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/App.tsx]: can you follow the same naming format that is being used, to diffrenetiate maybe you can add your name as prefix while importing let us not have _ while naming component

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/Dashboard.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import Feature3 from "./components/Feature3/Feature3";
+import Feature4 from "./components/Feature4/Feature4";
+import Feature5 from "./components/Feature5/Feature5";
+
+const Dashboard: React.FC = () => {
+  return (
+    <DashboardWrapper>
+      <div className="dashboard-container">
+          <nav className="navbar">
+            <ul className="feature-button">
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/Dashboard.tsx]: can you use a map for this instead of having multiple li elements also please add key for each li elements

File: interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/index.tsx [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import Dashboard from './Dashboard';
+
+const Week_4 = () => {
+  return (
+  <Dashboard/>
```
Comment (anusreesubash) [inline interns-react-weekly-assignments/src/components/Home/Shreyansh/Week_4/index.tsx]: is this file formatted?

File: src/components/Home/Shreyansh/Week_4/components/Feature1/Feature1Style.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import { styled } from 'styled-components';
+
+export const Feature1Wrapper = styled.div`
+  .funFactContainer {
```
Comment (RitikaGarg-ITT) [inline src/components/Home/Shreyansh/Week_4/components/Feature1/Feature1Style.ts]: you are using different naming convention in one file you are using hyphen naming in other file you are using camelCasing

File: src/components/Home/Shreyansh/index.tsx [hunk 13]
```diff
@@ -1,10 +1,24 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
 import { ShreyanshWrapper } from './styles';
 
 const Shreyansh = () => {
+  const [selectedWeek, setSelectedWeek] = useState('');
```
Comment (shreegovinda) [inline src/components/Home/Shreyansh/index.tsx:6]: type

File: src/components/Home/Keshav/Week1/index.tsx [hunk 14]
```diff
@@ -3,7 +3,7 @@ import { Week1Wrapper } from './styles';
 import ToDoList from './ToDoList/ToDoList';
 import { ToDoProvider } from './context/ToDoContext';
 
-const Week1: React.FC = () => {
+const Week1 = () => {
```
Comment (shreegovinda) [inline src/components/Home/Keshav/Week1/index.tsx]: why type is removed?

File: src/components/Home/Shreyansh/Week_4/components/Feature1/Feature1.tsx [hunk 15]
```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react';
+import { Feature1Wrapper } from './Feature1Style';
+import { APIKEY } from './utils';
+
+const Feature1: React.FC = () => {
+  const [fact, setFact] = useState<string>('');
+
+  const fetchRandomFact = async () => {
```
Comment (shreegovinda) [inline src/components/Home/Shreyansh/Week_4/components/Feature1/Feature1.tsx]: type

File: src/components/Home/Shreyansh/Week_4/Dashboard.tsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+              </Link>
+            </li>
+            <li>
+              <Link to='feature4'>
+                <button>Feature 4</button>
+              </Link>
+            </li>
+            <li>
+              <Link to='feature5'>
+                <button>Feature 5</button>
```
Comment (aditya-s-19) [inline src/components/Home/Shreyansh/Week_4/Dashboard.tsx]: See if these can be made concise with map()

File: src/components/Home/Shreyansh/Week_4/DashboardStyle.ts [hunk 17]
```diff
@@ -0,0 +1,46 @@
+import { styled } from 'styled-components';
+
+export const DashboardWrapper = styled.div`
+  .dashboard-container {
+    max-width: 800px;
+    margin: 30px auto 0;
+    padding: 20px;
+    background-color: gray;
```
Comment (aditya-s-19) [inline src/components/Home/Shreyansh/Week_4/DashboardStyle.ts]: both direct colors and hex values in code

File: src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+        <div>
+          <label>Amount:</label>
+          <input id='currencyInput' type='number' value={amount} onChange={handleAmountChange} />
+        </div>
+        <div>
+          <label>From:</label>
+          <select value={baseCurrency} onChange={handleBaseCurrencyChange}>
+            <option value='USD'>USD</option>
+            <option value='EUR'>EUR</option>
+            <option value='INR'>INR</option>
```
Comment (aditya-s-19) [inline src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx]: again, do it with map()

File: src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+  useEffect(() => {
+    const fetchExchangeRate = async () => {
+      const response = await fetch(`${APIKEY}${baseCurrency}`);
+      const data = await response.json();
+      setExchangeRate(data.rates[targetCurrency]);
+    };
+    fetchExchangeRate();
+  }, [baseCurrency, targetCurrency]);
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (aditya-s-19) [inline src/components/Home/Shreyansh/Week_4/components/Feature3/Feature3.tsx]: specify return types in functions as well

**Review by shreegovinda — DISMISSED** [review]

**Review by choudhary-keshav — DISMISSED** [review]

**Review by Harrrrry — DISMISSED** [review]

**Review by aditya-s-19 — DISMISSED** [review]

**Review by Harrrrry — APPROVED** [review]

**Review by aditya-s-19 — APPROVED** [review]

**Review by shreegovinda — APPROVED** [review]
